src/Compilers/CSharp/Test/Semantic/Semantics/MethodTypeInferenceTests.cs

```diff
@@ -633,6 +635,251 @@ static void Main(string[] args)
+    }
+   
+    static void TestA<T>(IOperation<T> operation, [MaybeNull] out T result) => result = default;
+    static void TestB<T>(IOperation<T> operation, out T result) => result = default;
+    
+    static void TestC<T>([MaybeNull] out T result) => result = default;
+    static void TestD<T>(out T result) => result = default;
+
+    static void TestE([MaybeNull] out string result) => result = default;
+    static void TestF(out string result) => result = "";
+}
+""";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.Net70);
+            comp.VerifyDiagnostics(new[] {
+                // (45,77): warning CS8601: Possible null reference assignment.
+                //     static void TestB<T>(IOperation<T> operation, out T result) => result = default;
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "default").WithLocation(45, 77),
+                // (48,52): warning CS8601: Possible null reference assignment.
+                //     static void TestD<T>(out T result) => result = default;
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "default").WithLocation(48, 52)
+            });
+
+            var tree = comp.SyntaxTrees.Single();
+            var model = comp.GetSemanticModel(tree);
+
+            // out _
+            foreach (var discardOut in GetDiscardIdentifiers(tree))
+            {
+                checkDiscard(model, discardOut, "System.String?");
+            }
+
+            // out T _, out var _, out T? _
+            foreach (var discardDecl in GetDiscardDesignations(tree))
+            {
+                checkDiscard(model, discardDecl, "System.String?");
+            }
+        }
+
+        [Fact]
+        public void Bug50782_2()
+        {
+            string source = """
+#nullable enable
+
+interface IOperation<T> 
+{ }
+
+public class StringOperation : IOperation<string> 
+{ }
+
+public class C 
+{   
+    static void Main() 
+    {
+        TestA(new StringOperation(), out string discardA);
+        TestA(new StringOperation(), out string _);
+        TestA(new StringOperation(), out var _);
+        TestA(new StringOperation(), out _);
+
+        TestB(out string discardB);
+        TestB(out string _);
+        TestB<string>(out var discardVarB);
+        TestB<string>(out var _);
+        TestB<string>(out _);
+
+        TestC(out var _);
+        TestC(out _);
+    }
+   
+    static void TestA<T>(IOperation<T> operation, out T result) => result = default;
+    static void TestB<T>(out T result) => result = default;
+    static void TestC(out string result) => result = "";
+}
+""";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.Net70);
+            comp.VerifyDiagnostics(new[] {                    
+                // (25,77): warning CS8601: Possible null reference assignment.
+                //     static void TestA<T>(IOperation<T> operation, out T result) => result = default;
```

Comment:
Consider assigning `default!` or similar since these warnings aren't really the interesting part of the test.